cmake/FindPyQt5.cmake

```diff
@@ -11,40 +11,53 @@
 #
 # This file defines the following variables:
 #
-# PYQT5_VERSION - The version of PyQt5 found expressed as a 6 digit hex number
-#     suitable for comparison as a string
-#
 # PYQT5_VERSION_STR - The version of PyQt5 as a human readable string.
 #
-# PYQT5_VERSION_TAG - The PyQt version tag using by PyQt's sip files.
-#
 # PYQT5_SIP_DIR - The directory holding the PyQt5 .sip files.
 #
 # PYQT5_SIP_FLAGS - The SIP flags used to build PyQt.
 
-IF(EXISTS PYQT5_VERSION)
+IF(EXISTS PYQT5_VERSION_STR)
   # Already in cache, be silent
   SET(PYQT5_FOUND TRUE)
-ELSE(EXISTS PYQT5_VERSION)
-
-  FIND_FILE(_find_pyqt5_py FindPyQt5.py PATHS ${CMAKE_MODULE_PATH} NO_CMAKE_FIND_ROOT_PATH)
-
-  EXECUTE_PROCESS(COMMAND ${Python_EXECUTABLE} ${_find_pyqt5_py} OUTPUT_VARIABLE pyqt_config)
-  IF(pyqt_config)
-    STRING(REGEX REPLACE "^pyqt_version:([^\n]+).*$" "\\1" PYQT5_VERSION ${pyqt_config})
-    STRING(REGEX REPLACE ".*\npyqt_version_str:([^\n]+).*$" "\\1" PYQT5_VERSION_STR ${pyqt_config})
-    STRING(REGEX REPLACE ".*\npyqt_version_tag:([^\n]+).*$" "\\1" PYQT5_VERSION_TAG ${pyqt_config})
-    STRING(REGEX REPLACE ".*\npyqt_version_num:([^\n]+).*$" "\\1" PYQT5_VERSION_NUM ${pyqt_config})
-    STRING(REGEX REPLACE ".*\npyqt_mod_dir:([^\n]+).*$" "\\1" PYQT5_MOD_DIR ${pyqt_config})
-    STRING(REGEX REPLACE ".*\npyqt_sip_dir:([^\n]+).*$" "\\1" PYQT5_SIP_DIR ${pyqt_config})
-    IF(EXISTS ${PYQT5_SIP_DIR}/Qt5)
-      SET(PYQT5_SIP_DIR ${PYQT5_SIP_DIR}/Qt5)
-    ENDIF(EXISTS ${PYQT5_SIP_DIR}/Qt5)
-    STRING(REGEX REPLACE ".*\npyqt_sip_flags:([^\n]+).*$" "\\1" PYQT5_SIP_FLAGS ${pyqt_config})
-    STRING(REGEX REPLACE ".*\npyqt_bin_dir:([^\n]+).*$" "\\1" PYQT5_BIN_DIR ${pyqt_config})
-    STRING(REGEX REPLACE ".*\npyqt_sip_module:([^\n]+).*$" "\\1" PYQT5_SIP_IMPORT ${pyqt_config})
-    SET(PYQT5_FOUND TRUE)
-  ENDIF(pyqt_config)
+ELSE(EXISTS PYQT5_VERSION_STR)
+
+  IF(SIP_BUILD_EXECUTABLE)
+    # SIP >= 5.0 path
+
+    FILE(GLOB _pyqt5_metadata "${Python_SITEARCH}/PyQt5*.dist-info/METADATA")
```

Comment:
maybe we are the only ones to do this... but it is not mandatory to have the dist-info when installing PyQt. There is a "--no-dist-info" which is enabled for our packages...

Maybe you can switch to the FindPyQt5 python script for this case, but I don't think it's useful, since you can build (and run!) QGIS if you have defined PYQT5_VERSION_STR and PYQT5_SIP_DIR

Reply:
good for me. Thanks